src/group_theory/submonoid/basic.lean

```diff
@@ -45,7 +44,7 @@ Note that `submonoid M` does not actually require `monoid M`, instead requiring
 `mul_one_class M`.
 
 This file is designed to have very few dependencies. In particular, it should not use natural
-numbers.
+numbers. It relies on an underlying subsemigroup.
```

Comment:
I find this sentence a little confusing (but it's maybe my fault). I would say "`submonoid` extends `subsemigroup` requiring `one_mem'`" or something like that.